src/ape/api/providers.py

```diff
@@ -719,3 +737,188 @@ def connection_str(self) -> str:
         The str used by downstream providers to connect to this one.
         For example, the URL for HTTP-based providers.
         """
+
+
+class SubprocessProvider(ProviderAPI):
+    """
+    A provider that manages a process, such as for ``ganache``.
+    """
+
+    PROCESS_WAIT_TIMEOUT = 15
+    process: Optional[Popen] = None
+    is_stopping: bool = False
+
+    @property
+    @abstractmethod
+    def process_name(self) -> str:
+        """The name of the process, such as ``Hardhat node``."""
```

Comment:
Is this note correct?

Reply:
Or wait, is this just used for display purposes? or calling subprocess?

Reply:
Yes this is just for display purposes, like the log statements `"Starting blank"`.

Perhaps there is a better way? hmmmm

The actual command comes from the abstract method `build_command()`.
Both `Hardhat` providers use this hook as needed, like the `mainnet-fork` provider includes the `--fork` and `--block-number` flags.

Reply:
Oh, and the first command in `ape-hardhat` is the `npx` bin path, so I can't find a reliable way to know the display name of a process based on its command list

Reply:
Okay, was confused for a sec lol